node_modules/oae-principals/lib/libraries/members.js

```diff
@@ -0,0 +1,293 @@
+ * such small latency if the members library hasn't been updated
+ */
+PrincipalsEmitter.on(PrincipalsConstants.events.LEFT_GROUP, function(ctx, group, leaveRole) {
+    var entry = {
+        'id': group.id,
+        'rank': _getMembersLibraryRank(group.id, ctx.user(), leaveRole),
+        'resource': ctx.user()
+    };
+
+    LibraryAPI.Index.remove(PrincipalsConstants.library.MEMBERS_INDEX_NAME, [entry]);
+});
+
+
+////////////////////////
+// INTERNAL FUNCTIONS //
+////////////////////////
+
+/**
+ * Get the numeric rank of a principal given the principal and their role in the members library
+ * they appear
+ *
+ * @param  {String}         groupId     The id of the group whose members library the principal is being inserted
+ * @param  {User|Group}     principal   The user or group for which to get their members library rank
+ * @param  {String}         role        The role for the principal in the members library
+ * @return {Number}                     The numeric rank to use for the principal in a members list
+ * @api private
+ */
+var _getMembersLibraryRank = function(groupId, principal, role) {
+    var libraryTenantAlias = AuthzUtil.getResourceFromId(groupId).tenantAlias;
+
+    // The picture visibility rank is based on if they have a profile picture, and how widely it
+    // is available to other users who might see the group
+    var pictureVisibilityRank = '0';
+
+    // The link visibility rank is based on how widely available their profile is to be viewed by
+    // other users. If a profile is not able to be "clicked", we don't want to rank them high in
+    // members displays
+    var linkVisibilityRank = '0';
+
+    // Lastly, managers who have wide visibility settings should be ranked higher than regular
+    // members with wide visibility settings
+    var roleRank = '0';
+
+    // A user's link visibility is directly related to their visibility leniency relative to the
+    // group's tenant
+    if (principal.visibility === AuthzConstants.visibility.PUBLIC) {
+        linkVisibilityRank = '2';
+    } else if (principal.visibility === AuthzConstants.visibility.LOGGEDIN && principal.tenant.alias === libraryTenantAlias) {
+        linkVisibilityRank = '1';
+    }
+
+    // A user's profile picture visibility ranking is only applicable if they have uploaded one. If
+    // so, it's visibility is identical to that of the profile link
+    if (_hasProfilePicture(principal)) {
+        pictureVisibilityRank = linkVisibilityRank;
+    }
+
+    // Simply rank managers higher
+    if (role === PrincipalsConstants.roles.MANAGER) {
+        roleRank = '1';
+    }
+
+    // Picture visibility (which implies link visibility) is the most important, secondary
+    // importance given to their link visibility. Finally, all else being equal, we prefer manager
+    // profiles over members
+    return util.format('%s%s%s', pictureVisibilityRank, linkVisibilityRank, roleRank);
```

Comment:
If it contained a `#` then yes, but I don't see how a rank of the form `000` would trip it up